### PR TITLE
Recipe edit

### PR DESCRIPTION
**Pull request by hari387**
Includes the following:
- Synchronous `RecipeServlet.doPut` which edits a recipe
- Update to `RecipeServlet.doPost` to make it synchronous using `ApiFuture`
- on recipe submit, redirect to newly submitted recipe

### CODE REVIEW

**Comment by hari387** [comment]
Closes #2 and #22

File: src/main/java/com/google/sps/servlets/RecipeServlet.java [hunk 1]
```diff
@@ -77,11 +77,37 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
 
     response.setContentType("application/json");
     response.getWriter().println(gson.toJson(new DBObject(newRecipe.id)));
-    response.setStatus(HttpServletResponse.SC_ACCEPTED);
+  }
+
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String data = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+    Recipe newRecipe = gson.fromJson(data, Recipe.class);
```
Comment (bradleypickard) [inline src/main/java/com/google/sps/servlets/RecipeServlet.java:96]: Where will this be fetched on the frontend? Are we planning to create a separate page to submit recipe edits?

Comment (hari387) [reply src/main/java/com/google/sps/servlets/RecipeServlet.java:96]: I would think so

Comment (nonsensicle) [reply src/main/java/com/google/sps/servlets/RecipeServlet.java:95]: I think we can include something on the profile page that displays recipes a user has submitted (by redirecting to recipe-display), and put the Edit option on those pages

File: src/main/webapp/submit-recipe.html [hunk 2]
```diff
@@ -57,13 +57,14 @@ <h1 id="title-preview"></h1>
...
-        }).then((response) => {
-          if (response.status === 202) {
-            window.location.href = "/feed.html"
-          } else {   
-            alert("Error " + response.status.toString())
-          }
         })
+          .then((response) => {
+            if (response.status !== 200) {
+              alert("Error " + response.status.toString())
```
Comment (nonsensicle) [inline src/main/webapp/submit-recipe.html:63]: Do we want to return here just so in the case of an error we dont end up having problems with response.json()?

Comment (hari387) [reply src/main/webapp/submit-recipe.html:63]: Yup, makes sense

**Pull request merged**